Document Blazor RenderTreeFrame breaking change (#20597)
dotnet/docs: includes/core-changes/aspnetcore/5.0/blazor-rendertreeframe-fields-become-properties.md

For discussion, see GitHub issue [dotnet/aspnetcore#25727](https://github.com/dotnet/aspnetcore/issues/25727).

#### Version introduced

5.0 RC1

#### Old behavior

Public members on `RenderTreeFrame` are defined as fields. For example, `renderTreeFrame.Sequence` and `renderTreeFrame.ElementName`.

#### New behavior

Public members on `RenderTreeFrame` are defined as properties with the same names as before. For example, `renderTreeFrame.Sequence` and `renderTreeFrame.ElementName`.

If older precompiled code hasn't been recompiled since this change, it may throw an exception similar to *MissingFieldException: Field not found: 'Microsoft.AspNetCore.Components.RenderTree.RenderTreeFrame.FrameType'*.

#### Reason for change

This change was necessary to implement high-impact performance improvements in Blazor component rendering in ASP.NET Core 5.0. The same levels of safety and encapsulation are maintained.

#### Recommended action

> Most Blazor developers are unaffected by this change. The change is more likely to affect library/package authors, but only in rare cases. Specifically, if you're developing:

Reviewer: ```suggestion
Most Blazor developers are unaffected by this change. The change is more likely to affect library and package authors, but only in rare cases. Specifically, if you're developing:
```

> * An app and using ASP.NET Core 3.x or upgrading to 5.0 RC1 or later, you don't need to change your own code. However, if you depend on a library that needs to upgrade to account for this change, then you need to update to a newer version of that library.

Reviewer: ```suggestion
* An app and using ASP.NET Core 3.x or upgrading to 5.0 RC1 or later, you don't need to change your own code. However, if you depend on a library that upgraded to account for this change, then you need to update to a newer version of that library.
```

> * A library and want to support only ASP.NET Core 5.0 RC1 or later, no action is needed. Just ensure that your project file declares the `<TargetFramework>` value of `net5.0`.

Reviewer: ```suggestion
* A library and want to support only ASP.NET Core 5.0 RC1 or later, no action is needed. Just ensure that your project file declares a `<TargetFramework>` value of `net5.0` or a later version.
```
